Source/Parallelization/WarpXComm.cpp

```diff
@@ -43,13 +43,18 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
     }
 #endif
 
-    const amrex::IntVect& Bx_stag = Bfield_fp[0][0]->ixType().toIntVect();
-    const amrex::IntVect& By_stag = Bfield_fp[0][1]->ixType().toIntVect();
-    const amrex::IntVect& Bz_stag = Bfield_fp[0][2]->ixType().toIntVect();
+    amrex::Vector<std::array<std::unique_ptr<amrex::MultiFab>,3>> const & Bmf = WarpX::fft_do_time_averaging ?
+                                                                                Bfield_avg_fp : Bfield_fp;
+    amrex::Vector<std::array<std::unique_ptr<amrex::MultiFab>,3>> const & Emf = WarpX::fft_do_time_averaging ?
+                                                                                Efield_avg_fp : Efield_fp;
```

Comment:
Without averaging, interpolate from `fp` to `aux`.
With averaging, interpolate from `avg_fp` to `aux`, which shares its data with `avg_aux`.

Reply:
This means in particular that the average fields are used for the diagnostics when `aux` is nodal (e.g. with momentum conserving field gather) but not when `aux` is not nodal (e.g. with momentum conserving field gather), right?

I noticed this when trying this branch because I'm using the option to compute the self fields at the beginning of the simulation. If I use energy conserving the self-fields are dumped at timestep 0 while if I use momentum conserving (+ time_averaging) the self-fields are not dumped at timestep 0. I don't think this is a serious issue but it might look like a bug to a user (e.g. "is the calculation of the self-field actually working if they're no longer in the output at timestep 0?"). It could be worth, e.g., updating the `avg_fp` arrays once the calculation of the self-fields has been performed, so that they appear in the diags at timestep 0.

Reply:
Hi Neal, I will reach out to you on Slack to discuss more details about this.